starter-code/app.js

```diff
@@ -1,18 +1,70 @@
-require('dotenv').config();
+require("dotenv").config();
 
-const express = require('express');
-const hbs = require('hbs');
+const express = require("express");
+const hbs = require("hbs");
 
 // require spotify-web-api-node package here:
+const SpotifyWebApi = require("spotify-web-api-node");
 
 const app = express();
 
-app.set('view engine', 'hbs');
-app.set('views', __dirname + '/views');
-app.use(express.static(__dirname + '/public'));
+app.set("view engine", "hbs");
+app.set("views", __dirname + "/views");
+app.use(express.static(__dirname + "/public"));
+hbs.registerPartials(__dirname + "/views/partials");
 
 // setting the spotify-api goes here:
+const spotifyApi = new SpotifyWebApi({
+  clientId: process.env.CLIENT_ID,
+  clientSecret: process.env.CLIENT_SECRET
+});
+
+// Retrieve an access token
+spotifyApi
+  .clientCredentialsGrant()
+  .then(data => spotifyApi.setAccessToken(data.body["access_token"]))
+  .catch(error =>
+    console.log("Something went wrong when retrieving an access token", error)
+  );
 
 // Our routes go here:
 
-app.listen(3000, () => console.log('My Spotify project running on port 3000 🎧 🥁 🎸 🔊'));
+app.listen(3000, () =>
+  console.log("My Spotify project running on port 3000 🎧 🥁 🎸 🔊")
+);
+
+app.get("/", (req, res) => {
+  res.render("index.hbs", {});
+});
+app.get("/artist-search", (req, res) => {
+  spotifyApi
+    .searchArtists(req.query.artist)
+    .then(data => {
+      res.render("artist-search-results.hbs", {
+        artists: data.body.artists.items
+      });
+    })
+    .catch(err =>
+      console.log("The error while searching artists occurred: ", err)
+    );
+});
+app.get("/albums", (req, res, next) => {
```

Comment:
Here it is better to use: app.get("/albums/:id", .....

and then use req.params.id    this will give you back the id and it looks better. req.query is meant for the search function, while req.params are neater for getting a particular item that is clicked.